flocker/provision/_install.py

```diff
@@ -316,48 +315,85 @@ def install_cli_commands_ubuntu(distribution, package_source):
             Package:  *
             Pin: origin {}
             Pin-Priority: 900
-            '''.format(buildbot_host)), '/tmp/apt-pref'))
-        commands.append(sudo_from_args([
+        '''.format(buildbot_host)), '/tmp/apt-pref'))
+        commands.append(run_from_args([
             'mv', '/tmp/apt-pref', '/etc/apt/preferences.d/buildbot-900']))
 
     # Update to read package info from new repos
-    commands.append(sudo_from_args(["apt-get", "update"]))
+    commands.append(run_from_args(["apt-get", "update"]))
 
     if package_source.os_version:
-        package = 'clusterhq-flocker-cli=%s' % (package_source.os_version,)
-    else:
-        package = 'clusterhq-flocker-cli'
+        package_name += '=%s' % (package_source.os_version,)
 
-    # Install Flocker CLI and all dependencies
-    commands.append(sudo_from_args([
-        'apt-get', '-y', '--force-yes', 'install', package]))
+    # Install package and all dependencies
+    commands.append(run_from_args([
+        'apt-get', '-y', '--force-yes', 'install', package_name]))
 
     return sequence(commands)
 
 
 _task_install_commands = {
-    'centos-7': install_cli_commands_yum,
-    'ubuntu-14.04': install_cli_commands_ubuntu,
-    'ubuntu-15.04': install_cli_commands_ubuntu,
+    'centos-7': install_commands_yum,
+    'ubuntu-14.04': install_commands_ubuntu,
+    'ubuntu-15.04': install_commands_ubuntu,
```

Comment:
Keeping this mapping and adding the `is_centos/ubuntu` functions means that there are now multiple ways to convert between a specific distribution and a base system (analogous to supported packaging system).  It would be better to have a single source of truth for this.  See https://clusterhq.atlassian.net/browse/FLOC-1670 for some ideas.